src/hooks/Guilds/guild/useFilteredProposalActions.tsx

```diff
@@ -0,0 +1,87 @@
+import { useMemo } from 'react';
+import useProposalCalls from 'hooks/Guilds/guild/useProposalCalls';
+import {
+  DecodedAction,
+  SupportedAction,
+} from 'old-components/Guilds/ActionsBuilder/types';
+
+const isApprovalCall = (action: DecodedAction) => {
+  // TODO: improve this with better conditions. At read point decodedAction has no other things to validate than generic call && approval event in contract interface
+  const type = action?.decodedCall?.callType;
+  return (
+    type === SupportedAction.GENERIC_CALL &&
+    Object.keys(action?.contract?.events ?? {}).some(key =>
+      key.includes('Approval')
+    )
+  );
+};
+
+/**
+ * Importance:
+ * 1. rep minting
+ * 2. spending calls
+ * 3. transfers.
+ * 4. generic calls
+ */
+const getActionPoints = (action: DecodedAction): number => {

```

Comment:
I have a hunch we might forget to update this when we introduce new SupportedAction types later. Shall we move this importance value to src/old-components/Guilds/ActionsBuilder/SupportedActions/index.tsx. We should also update the TypeScript typings to make it required.

Reply:
I moved this fn to the supported actions file to hold this pointing logic. I don't know if I can add types for this unless the approval call is a supported action.  Make sense?